lib/charms/operator_libs_linux/v0/juju_systemd_notices.py

```diff
@@ -256,6 +292,16 @@ def _dbus_path_to_name(path: str) -> str:
     return name
 
 
+@functools.lru_cache(maxsize=32)
+def _read_config() -> Mapping[str, str]:
+    """Read systemd notices daemon configuration to service names and aliases."""
+    config_file = Path.cwd() / "watch.yaml"
+    _logger.debug("Loading observed services from configuration file %s", config_file)
+
+    with config_file.open("rt") as fin:
+        return yaml.safe_load(fin)["services"]
```

Comment:
I do like the more explicit way of telling the background process about the service names (and aliases). However, writing to a YAML config seems a bit heavy (and requires an addition to the PYDEPS). What about avoiding the YAML dependency and extra complexity by using command line args:

```
services = [f'{s.name}={s.alias}' if s.alias else s.name for s in self._services]
service_file.write_text(f"""
...
ExecStart=/usr/bin/python3 {__file__} {self._charm.unit.name} {' '.join(services)}
...
""")
```

And then the main() would use argparse with nargs='*' (I think it is) and read them from the CLI argument.

It may also make testing slightly easier.

Reply:
I like this suggestion. I'll test it out. Big thing is ensuring how we can "bubble down" the observed service configuration to the daemon's methods. Could likely use a global map similar to `_service_states` where we just set a map of the watched services when `argparse` parses the arguments passed to the daemon.

Reply:
I went ahead with this approach. Works nicely!